src/WebListener.js

```diff
@@ -53,26 +45,41 @@ export default class WebListener {
     const awaitingBindings = Promise.all(this._bindings.map(binding => this._send('bind', binding)));
 
     const isReady = new Promise((resolve, reject) => {
-      const once = () => {
-        this.off('ready', resolve);
-        this.off('error', reject);
-        this.off('close', reject);
+      const offReady = this.on('ready', resolve);
+      const offError = this.on('error', reject);
+      const offClose = this.on('close', reject);
+      const unbindAll = () => {
+        offReady();
+        offError();
+        offClose();
       };
 
-      this.on('ready', resolve);
-      this.on('ready', once);
-      this.on('error', reject);
-      this.on('error', once);
-      this.on('close', reject);
-      this.on('close', once);
+      this.on('ready', unbindAll);
+      this.on('error', unbindAll);
+      this.on('close', unbindAll);
     });
 
     // When all bindings have been bound, we're just waiting for 'ready'
-    return awaitingBindings.then(() => isReady);
+    return awaitingBindings
+      .then(() => {
+        this.connectInterval = setInterval(() => {
+          if (!this.socket || this.socket.readyState === WebSocket.CLOSED) {
+            this.connect();
+          }
+        }, this.options.reconnectInterval);
+
+        return isReady;
+      });
+  }
+
+  reconnect() {
+    if (!this.socket || this.socket.readyState === WebSocket.CLOSED) {
+      this.connect();
```

Comment:
I would suggest emitting 'reconnected' after a successful reconnect...

In task inspector we should ideally reload and synchronize state when reconnect is emitted...

Reply:
I'm not too keen on doing a full reload, but I'm not sure we have much of a choice.